When targetting i686-pc-windows-msvc linker fails

When I create a binary targetting `x86_64-pc-windows-msvc` everything seems to work as expected. When I change to target `i686-pc-windows-msvc` I get a linking error:

```
error: linking with `link.exe` failed: exit code: 1120
  |
  = note: "C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\bin\\amd64_x86\\link.exe" "/NOLOGO" "/NXCOMPAT" "/LARGEADDRESSAWARE" "/SAFESEH" "/LIBPATH:C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\grpctestclient-22dd802a760831fb.0.o" "/OUT:C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\grpctestclient-22dd802a760831fb.exe" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\grpctestclient-22dd802a760831fb.crate.allocator.o" "/OPT:REF,NOICF" "/DEBUG" "/LIBPATH:C:\\code\\rust\\grpctestclient\\target\\debug\\deps" "/LIBPATH:C:\\code\\rust\\grpctestclient\\target\\debug\\build\\grpcio-sys-8d76f4e474ce88cb\\out/build/Debug" "/LIBPATH:C:\\code\\rust\\grpctestclient\\target\\debug\\build\\grpcio-sys-8d76f4e474ce88cb\\out/build/third_party/cares/Debug" "/LIBPATH:C:\\code\\rust\\grpctestclient\\target\\debug\\build\\grpcio-sys-8d76f4e474ce88cb\\out/build/third_party/zlib/Debug" "/LIBPATH:C:\\code\\rust\\grpctestclient\\target\\debug\\build\\grpcio-sys-8d76f4e474ce88cb\\out/build/third_party/boringssl/ssl/Debug" "/LIBPATH:C:\\code\\rust\\grpctestclient\\target\\debug\\build\\grpcio-sys-8d76f4e474ce88cb\\out/build/third_party/boringssl/crypto/Debug" "/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\atlmfc\\lib\\" "/LIBPATH:C:\\code\\rust\\grpctestclient\\target\\debug\\build\\grpcio-sys-8d76f4e474ce88cb\\out" "/LIBPATH:C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libtest-4a2ffbd5d1de8748.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libgetopts-bb18395838510ee5.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libgrpcio-d135e1252312e2fe.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libterm-e82ad60c07d5ce89.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libtls_api_native_tls-4d98a338377eb70b.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libgrpcio_sys-c1500af703d76486.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libnative_tls-dd77dac0d4b4d2f9.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\liblibc-c819602701b7e0ca.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libschannel-db045e030030947e.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\liblazy_static-a8ae7b6260dfbfb6.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libadvapi32-c5f0c7faeb05dd68.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libprotobuf-8f6e9a787c870112.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libcrypt32-bfac4fca0bf3c582.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libhttpbis-5698fdf2a284a85d.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libtokio_tls_api-7005842da04e624b.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libtls_api_stub-baa0617cf4118d4c.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libvoid-e98405aa43567aa9.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libtokio_timer-1ca6361aeb5dea75.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libtokio_core-d6137ee4a21ff9d3.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libscoped_tls-0c128279b6f54e76.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libtokio_io-0a452ec1c3e89601.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libmio-af66d2fa429b98ee.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libmiow-efba840c83ef764a.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libslab-3682ad05451db221.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\liblazycell-d93b41e1e6cfc529.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libbytes-a293a0e693d816bf.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libiovec-7342dacdd18faf24.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libbyteorder-9f659a19469e04e7.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\liblog-cbb1b7356a5089ec.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libsecur32-0b426bc72f20236e.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libtls_api-05bc1aa0754fce7d.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libnet2-c2616e29d4ce9af8.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libcfg_if-342e0d390982db09.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libws2_32-8a8a075799b13f18.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libkernel32-c15f9a2439315d77.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libwinapi-8b24607fe36df8d1.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libfutures_cpupool-bb627e48a89392d4.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libnum_cpus-6f264736150b20e1.rlib" "C:\\code\\rust\\grpctestclient\\target\\debug\\deps\\libfutures-c07957cef9ac0120.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libstd-cb715f4af2c20b97.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libpanic_unwind-0fa8dbd47e2c76ee.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libunwind-c591e8b5c7fa4410.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\liblibc-4a73c6ed200bc6a7.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\liballoc_system-149ec0f4aeac373b.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\liballoc-7ced75ab424a1846.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libstd_unicode-11a286598ffea675.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\librand-093396b7aafa4528.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libcore-8b3b509fc1c5b3f9.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libcompiler_builtins-e9a8d8f9e3a2e5ce.rlib" "kernel32.lib" "ncrypt.lib" "advapi32.lib" "crypt32.lib" "secur32.lib" "ws2_32.lib" "kernel32.lib" "advapi32.lib" "ws2_32.lib" "userenv.lib" "shell32.lib" "msvcrt.lib"
  = note: grpctestclient-22dd802a760831fb.0.o : error LNK2019: unresolved external symbol _grpcwrap_call_start_duplex_streaming referenced in function __ZN6grpcio4call6client36_$LT$impl$u20$grpcio..call..Call$GT$16duplex_streaming28_$u7b$$u7b$closure$u7d$$u7d$17h1d7b3d1e0aec55cbE
          grpctestclient-22dd802a760831fb.0.o : error LNK2019: unresolved external symbol _grpcwrap_call_recv_initial_metadata referenced in function __ZN6grpcio4call6client36_$LT$impl$u20$grpcio..call..Call$GT$16duplex_streaming28_$u7b$$u7b$closure$u7d$$u7d$17ha4594e2bed7e0e93E
          grpctestclient-22dd802a760831fb.0.o : error LNK2019: unresolved external symbol _grpcwrap_channel_create_call referenced in function __ZN6grpcio7channel7Channel11create_call17h88b7521c5d488f9cE
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_request_call_context_create referenced in function __ZN6grpcio4call6server14RequestContext3new17hcd767579623c2bbbE
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_request_call_context_get_call referenced in function __ZN6grpcio4call6server14RequestContext16handle_unary_req17h9ed79af7c4e04ce1E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_call_recv_message referenced in function __ZN6grpcio4call6server14RequestContext16handle_unary_req17h9ed79af7c4e04ce1E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_request_call_context_take_call referenced in function __ZN6grpcio4call6server14RequestContext9take_call17h950537a2864000a3E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_request_call_context_method referenced in function __ZN6grpcio4call6server14RequestContext6method17h61867e52d772dc37E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_request_call_context_host referenced in function __ZN6grpcio4call6server14RequestContext4host17h36740be0b259d3d1E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_request_call_context_deadline referenced in function __ZN6grpcio4call6server14RequestContext8deadline17ha4e6407b18c20ac4E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_request_call_context_destroy referenced in function __ZN78_$LT$grpcio..call..server..RequestContext$u20$as$u20$core..ops..drop..Drop$GT$4drop17h6903b934e12ca739E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_batch_context_create referenced in function __ZN6grpcio4call12BatchContext3new17h313eb149fc0232d6E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_batch_context_recv_status_on_client_status referenced in function __ZN6grpcio4call12BatchContext10rpc_status17h19bf73b538dd97b6E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_batch_context_recv_status_on_client_details referenced in function __ZN6grpcio4call12BatchContext10rpc_status17h19bf73b538dd97b6E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_batch_context_recv_message_length referenced in function __ZN6grpcio4call12BatchContext12recv_message17hecad8a06d1154dc1E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_batch_context_recv_message_to_buffer referenced in function __ZN6grpcio4call12BatchContext12recv_message17hecad8a06d1154dc1E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_batch_context_destroy referenced in function __ZN68_$LT$grpcio..call..BatchContext$u20$as$u20$core..ops..drop..Drop$GT$4drop17h03cbf64e3116f2d2E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_call_send_message referenced in function __ZN6grpcio4call4Call18start_send_message28_$u7b$$u7b$closure$u7d$$u7d$17h10144ea50e6d50ebE
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_call_send_close_from_client referenced in function __ZN6grpcio4call4Call23start_send_close_client28_$u7b$$u7b$closure$u7d$$u7d$17h1d578467004b386cE
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_call_start_serverside referenced in function __ZN6grpcio4call4Call17start_server_side28_$u7b$$u7b$closure$u7d$$u7d$17h8d7af00415d0a313E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_call_send_status_from_server referenced in function __ZN6grpcio4call4Call29start_send_status_from_server28_$u7b$$u7b$closure$u7d$$u7d$17h36ba0ae05577b75dE
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_channel_args_create referenced in function __ZN6grpcio7channel14ChannelBuilder10build_args17h205b65bf5e901e37E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_channel_args_set_integer referenced in function __ZN6grpcio7channel14ChannelBuilder10build_args17h205b65bf5e901e37E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_channel_args_set_string referenced in function __ZN6grpcio7channel14ChannelBuilder10build_args17h205b65bf5e901e37E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_channel_args_destroy referenced in function __ZN70_$LT$grpcio..channel..ChannelArgs$u20$as$u20$core..ops..drop..Drop$GT$4drop17h6414f3e9806d0052E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_ssl_server_credentials_create referenced in function __ZN6grpcio11credentials24ServerCredentialsBuilder5build17h9d376d731225cac4E
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_ssl_credentials_create referenced in function __ZN6grpcio11credentials25ChannelCredentialsBuilder5build17h120f4d402cfaa98fE
          libgrpcio-d135e1252312e2fe.rlib(grpcio-d135e1252312e2fe.0.o) : error LNK2019: unresolved external symbol _grpcwrap_server_request_call referenced in function __ZN6grpcio6server12request_call17h15fec5bc04c4d623E
          C:\code\rust\grpctestclient\target\debug\deps\grpctestclient-22dd802a760831fb.exe : fatal error LNK1120: 28 unresolved externals


error: aborting due to previous error
```

A similiar thing happens when running `cargo test` on `grpc-rc` targetting `i686-pc-windows-msvc` (again `x86_64-pc-windows-msvc` works as expected):

```
error: linking with `link.exe` failed: exit code: 1120
  |
  = note: "C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\bin\\amd64_x86\\link.exe" "/NOLOGO" "/NXCOMPAT" "/LARGEADDRESSAWARE" "/SAFESEH" "/LIBPATH:C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib" "C:\\code\\rust\\grpc-rs\\target\\debug\\examples\\greeter_server-5b7ff584f698d1c3.0.o" "/OUT:C:\\code\\rust\\grpc-rs\\target\\debug\\examples\\greeter_server-5b7ff584f698d1c3.exe" "C:\\code\\rust\\grpc-rs\\target\\debug\\examples\\greeter_server-5b7ff584f698d1c3.crate.allocator.o" "/OPT:REF,NOICF" "/DEBUG" "/LIBPATH:C:\\code\\rust\\grpc-rs\\target\\debug\\deps" "/LIBPATH:C:\\code\\rust\\grpc-rs\\target\\debug\\build\\grpcio-sys-ad7ef0ad3da8d23a\\out/build/Debug" "/LIBPATH:C:\\code\\rust\\grpc-rs\\target\\debug\\build\\grpcio-sys-ad7ef0ad3da8d23a\\out/build/third_party/cares/Debug" "/LIBPATH:C:\\code\\rust\\grpc-rs\\target\\debug\\build\\grpcio-sys-ad7ef0ad3da8d23a\\out/build/third_party/zlib/Debug" "/LIBPATH:C:\\code\\rust\\grpc-rs\\target\\debug\\build\\grpcio-sys-ad7ef0ad3da8d23a\\out/build/third_party/boringssl/ssl/Debug" "/LIBPATH:C:\\code\\rust\\grpc-rs\\target\\debug\\build\\grpcio-sys-ad7ef0ad3da8d23a\\out/build/third_party/boringssl/crypto/Debug" "/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\atlmfc\\lib\\" "/LIBPATH:C:\\code\\rust\\grpc-rs\\target\\debug\\build\\grpcio-sys-ad7ef0ad3da8d23a\\out" "/LIBPATH:C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libgrpcio_proto-f81184248277bc6e.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libslog_term-68b5471fa7f08fbf.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libterm-f854bf64fae3c501.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libisatty-e3e6fe9705e46736.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libchrono-62d9840c3b21e416.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libnum-c0ce31775841def5.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libtime-8fc3a3e9a37310e2.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libkernel32-c15f9a2439315d77.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libwinapi-8b24607fe36df8d1.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libslog_stdlog-7a53680d894706bd.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libslog_scope-7d3a96d7a1425a26.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libslog_async-7c99da4eb6b7296b.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libtake_mut-2ecaf07df722bece.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libthread_local-9ccb72a75948b8ef.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\liblazy_static-06b18921fa9d687f.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libunreachable-3ebc2be93b299b04.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libvoid-758ade381656fe45.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libgrpcio-5e40bbfcffa50f6a.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libprotobuf-8f6e9a787c870112.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\liblog-cbb1b7356a5089ec.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libgrpcio_sys-2989227e05ced6b1.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\liblibc-c819602701b7e0ca.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libfutures-c07957cef9ac0120.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libnum_iter-83c1e49d83d2a8e7.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libnum_integer-12644b41e0d481e7.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libnum_traits-89eaeeb6951b2c47.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libcrossbeam-b8befa5a22f932f8.rlib" "C:\\code\\rust\\grpc-rs\\target\\debug\\deps\\libslog-1420a44cbad7e71e.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libstd-cb715f4af2c20b97.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libpanic_unwind-0fa8dbd47e2c76ee.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libunwind-c591e8b5c7fa4410.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\liblibc-4a73c6ed200bc6a7.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\liballoc_system-149ec0f4aeac373b.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\liballoc-7ced75ab424a1846.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libstd_unicode-11a286598ffea675.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\librand-093396b7aafa4528.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libcore-8b3b509fc1c5b3f9.rlib" "C:\\Users\\Simon\\.rustup\\toolchains\\stable-i686-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libcompiler_builtins-e9a8d8f9e3a2e5ce.rlib" "kernel32.lib" "advapi32.lib" "ws2_32.lib" "userenv.lib" "shell32.lib" "msvcrt.lib"
  = note: libgrpcio_proto-f81184248277bc6e.rlib(grpcio_proto-f81184248277bc6e.0.o) : error LNK2019: unresolved external symbol _grpcwrap_call_start_unary referenced in function __ZN6grpcio4call6client36_$LT$impl$u20$grpcio..call..Call$GT$11unary_async28_$u7b$$u7b$closure$u7d$$u7d$17h03917d3e6fab05b3E
          libgrpcio_proto-f81184248277bc6e.rlib(grpcio_proto-f81184248277bc6e.0.o) : error LNK2019: unresolved external symbol _grpcwrap_call_start_client_streaming referenced in function __ZN6grpcio4call6client36_$LT$impl$u20$grpcio..call..Call$GT$16client_streaming28_$u7b$$u7b$closure$u7d$$u7d$17h10541628a09acfaeE
          libgrpcio_proto-f81184248277bc6e.rlib(grpcio_proto-f81184248277bc6e.0.o) : error LNK2019: unresolved external symbol _grpcwrap_call_start_duplex_streaming referenced in function __ZN6grpcio4call6client36_$LT$impl$u20$grpcio..call..Call$GT$16duplex_streaming28_$u7b$$u7b$closure$u7d$$u7d$17h0864579e6e590fc2E
          libgrpcio_proto-f81184248277bc6e.rlib(grpcio_proto-f81184248277bc6e.0.o) : error LNK2019: unresolved external symbol _grpcwrap_call_recv_initial_metadata referenced in function __ZN6grpcio4call6client36_$LT$impl$u20$grpcio..call..Call$GT$16duplex_streaming28_$u7b$$u7b$closure$u7d$$u7d$17h149b9bea1d749922E
          libgrpcio_proto-f81184248277bc6e.rlib(grpcio_proto-f81184248277bc6e.0.o) : error LNK2019: unresolved external symbol _grpcwrap_call_start_server_streaming referenced in function __ZN6grpcio4call6client36_$LT$impl$u20$grpcio..call..Call$GT$16server_streaming28_$u7b$$u7b$closure$u7d$$u7d$17h7940bf8c2089c134E
          libgrpcio_proto-f81184248277bc6e.rlib(grpcio_proto-f81184248277bc6e.0.o) : error LNK2019: unresolved external symbol _grpcwrap_channel_create_call referenced in function __ZN6grpcio7channel7Channel11create_call17h039d52431350efe7E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_request_call_context_create referenced in function __ZN6grpcio4call6server14RequestContext3new17hd12bf3663bccef6dE
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_request_call_context_get_call referenced in function __ZN6grpcio4call6server14RequestContext16handle_unary_req17hc07924e60a276058E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_call_recv_message referenced in function __ZN6grpcio4call6server14RequestContext16handle_unary_req17hc07924e60a276058E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_request_call_context_take_call referenced in function __ZN6grpcio4call6server14RequestContext9take_call17h8fdb087fb6eee43eE
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_request_call_context_method referenced in function __ZN6grpcio4call6server14RequestContext6method17hc8ef6f202af44022E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_request_call_context_host referenced in function __ZN6grpcio4call6server14RequestContext4host17h7383bd1de4aa4299E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_request_call_context_deadline referenced in function __ZN6grpcio4call6server14RequestContext8deadline17hf0949643c3dd3c85E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_request_call_context_destroy referenced in function __ZN78_$LT$grpcio..call..server..RequestContext$u20$as$u20$core..ops..drop..Drop$GT$4drop17hedee1af8d4b5d909E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_batch_context_create referenced in function __ZN6grpcio4call12BatchContext3new17h18bcd2e1798a9423E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_batch_context_recv_status_on_client_status referenced in function __ZN6grpcio4call12BatchContext10rpc_status17h924a64a7789c5d69E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_batch_context_recv_status_on_client_details referenced in function __ZN6grpcio4call12BatchContext10rpc_status17h924a64a7789c5d69E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_batch_context_recv_message_length referenced in function __ZN6grpcio4call12BatchContext12recv_message17h665a114a792cf18cE
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_batch_context_recv_message_to_buffer referenced in function __ZN6grpcio4call12BatchContext12recv_message17h665a114a792cf18cE
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_batch_context_destroy referenced in function __ZN68_$LT$grpcio..call..BatchContext$u20$as$u20$core..ops..drop..Drop$GT$4drop17haa599fad5b97746cE
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_call_send_message referenced in function __ZN6grpcio4call4Call18start_send_message28_$u7b$$u7b$closure$u7d$$u7d$17h3dcef4688cbaa526E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_call_send_close_from_client referenced in function __ZN6grpcio4call4Call23start_send_close_client28_$u7b$$u7b$closure$u7d$$u7d$17h0d979c57620bdf95E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_call_start_serverside referenced in function __ZN6grpcio4call4Call17start_server_side28_$u7b$$u7b$closure$u7d$$u7d$17h3f816abf173a115aE
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_call_send_status_from_server referenced in function __ZN6grpcio4call4Call29start_send_status_from_server28_$u7b$$u7b$closure$u7d$$u7d$17hcfb54c8319a74d4eE
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_channel_args_create referenced in function __ZN6grpcio7channel14ChannelBuilder10build_args17hb2cb25756f4e6b53E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_channel_args_set_integer referenced in function __ZN6grpcio7channel14ChannelBuilder10build_args17hb2cb25756f4e6b53E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_channel_args_set_string referenced in function __ZN6grpcio7channel14ChannelBuilder10build_args17hb2cb25756f4e6b53E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_channel_args_destroy referenced in function __ZN70_$LT$grpcio..channel..ChannelArgs$u20$as$u20$core..ops..drop..Drop$GT$4drop17hbe7b031a31eae858E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_ssl_server_credentials_create referenced in function __ZN6grpcio11credentials24ServerCredentialsBuilder5build17hdbd9f7d5cf0ce426E
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_ssl_credentials_create referenced in function __ZN6grpcio11credentials25ChannelCredentialsBuilder5build17haf98991752a8c69bE
          libgrpcio-5e40bbfcffa50f6a.rlib(grpcio-5e40bbfcffa50f6a.0.o) : error LNK2019: unresolved external symbol _grpcwrap_server_request_call referenced in function __ZN6grpcio6server12request_call17h1ccc10e8a779bbeeE
          C:\code\rust\grpc-rs\target\debug\examples\greeter_server-5b7ff584f698d1c3.exe : fatal error LNK1120: 31 unresolved externals


error: aborting due to previous error

error: Could not compile `grpcio`.
```

I tried running `cargo test` of `grpc-rc` on a clean Windows 10 32bit VM too to see if that would fix it, but I get the same error as above.

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -7,7 +7,9 @@ cache:
   - C:\ProgramData\chocolatey\lib -> appveyor.yml
 
 environment:
-  TARGET: x86_64-pc-windows-msvc
+  matrix:
+  - TARGET: x86_64-pc-windows-msvc
+  - TARGET: i686-pc-windows-msvc
 
 install:
   - ps: Start-FileDownload "https://static.rust-lang.org/dist/rust-nightly-${env:TARGET}.exe"

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -25,5 +25,5 @@ build: false
 
 test_script:
   - SET RUSTFLAGS=-Dwarnings
-  - cargo build
-  - cargo test --all
+  - cargo build --target %TARGET%
+  - cargo test --all --target %TARGET%

--- a/grpc-sys/grpc_wrap.c
+++ b/grpc-sys/grpc_wrap.c
@@ -62,7 +62,7 @@
 
 #ifdef GPR_WINDOWS
 #define GPR_EXPORT __declspec(dllexport)
-#define GPR_CALLTYPE __stdcall
+#define GPR_CALLTYPE __cdecl
 #endif
 
 #ifndef GPR_EXPORT

